src/bootstrap/dist.rs

```diff
@@ -2058,6 +2058,12 @@ impl Step for RustDev {
 
         builder.ensure(crate::native::Llvm { target });
 
+        // If the config has LLD enabled, ensure its step. We'll we want to package it, and use it
+        // in download-ci-llvm.
+        if builder.config.lld_enabled {
+            builder.ensure(crate::native::Lld { target });
```

Comment:
Shouldn't we do this regardless of whether lld is enabled? We want `RustDev` to be the same across all platforms.

Reply:
Do you mean the contents of the archive (as this is already not the same across all platforms IIUC) ?

Reply:
Yes, I mean the contents.

>  But as mentioned in the OP, this ensure is not strictly necessary, packaging lld is already done if it's found on disk.

Right, and I think that's very non-intuitive and we should change it. `lld` should be packaged unconditionally, not based on `lld_enabled` - that config says
> Indicates whether LLD will be compiled and made available in the sysroot for rustc to execute.

but IMO that should only affect whether LLD is *used* when building rustc, not whether we package it or not.

What would be the purpose of making this configurable? We'll want to make sure it's enabled in all dist builders anyway, and at that point it just becomes one more thing we need to test.

Reply:
> but IMO that should only affect whether LLD is _used_ when building rustc

There's another different setting for that, `rust.use-lld`. I assumed the purpose of `rust.lld` is to be able to choose the subset of platforms for which we distribute `rust-lld`, e.g. maybe it itself doesn't support some platforms we need, but I don't know either. 

If the comment `// We don't build LLD on some platforms, so only add it if it exists` is still true, it may not yet be possible to package `lld` unconditionally. We'll probably have to see what CI has to say about that :)

Reply:
god, what a mess :(

ok, we don't need to dig into the difference here. But I do still think `RustDev` should always have lld in it.